### ISSUE TITLE
ROOT does not work when launched from non-accessible folder

### ISSUE THREAD

**Issue by ktf** [root-project/root#12409]
- [x ] Checked for duplicates

### Describe the bug

ROOT 6.28.00 does not work correctly when $PWD is not accessible by the user.

If I do:

```bash
mkdir -p foo
cd foo
chmod 000 .
root
```

it fails with:

```bash
input_line_2:1:10: fatal error: cannot open file './cling/Interpreter/RuntimeUniverse.h': Permission denied
#include "cling/Interpreter/RuntimeUniverse.h"
         ^
Replaced symbol atexit cannot be found in JIT!
Replaced symbol at_quick_exit cannot be found in JIT!
input_line_4:2:10: fatal error: cannot open file './TError.h': Permission denied
#include "TError.h"
         ^
input_line_5:1:10: fatal error: cannot open file './cling/Interpreter/DynamicLookupRuntimeUniverse.h': Permission denied
#include "cling/Interpreter/DynamicLookupRuntimeUniverse.h"
```

the same snippet works fine with 6.26.10.

Notice things work fine with just the x bit set (i.e. chmod 100 . still works).

### Expected behavior

The standard root prompt without any error.

### To Reproduce

See above.

```bash
mkdir -p foo
cd foo
chmod 000 .
root
```

### Setup

ROOT 6.28.00, both el7 and ubuntu.

### Additional context

While this can look like a corner case, it reflects the way ALICE runs its online reconstruction jobs, so it's rather critical for us. I am in parallel trying to understand if the unaccessible cwd can be avoided, but this might have non trivial implications (and its for sure beyond my responsibilities).

**Comment by ktf** [comment]
@knopers @vascobarroso FYI

**Comment by ktf** [comment]
@barthelemy

**Comment by ktf** [comment]
I had a typo in the reproducer, now corrected.

**Comment by ktf** [comment]
For ALICE people see also https://alice.its.cern.ch/jira/browse/O2-3591.

**Comment by ktf** [comment]
FYI @pzhristov

**Comment by Axel-Naumann** [comment]
Update: I can reproduce this with the (for our tests nicer) case
```bash
mkdir cling
chmod a-xr cling
root
```
which gives the same diagnostics as above.

**Comment by Axel-Naumann** [comment]
clang does not work when launched from non-accessible folder, either:
```
$ mkdir noread
$ cd noread/
$ chmod a-rwx .
$ echo '#include "ROOT/config/RVersion.hxx"' | clang++ -I. -I$ROOTSYS/include/  -fsyntax-only -x c++ -
<stdin>:1:10: fatal error: cannot open file './ROOT/config/RVersion.hxx': Permission denied
#include "ROOT/config/RVersion.hxx"
         ^
1 error generated.
```

I.e. clang tries to argue "don't do that". But I would argue "clang, please." because it even just goes on and finds the header!

This was introduced [here](https://github.com/llvm/llvm-project/commit/babdfdec90bb978799bcccf1ee0c856678b0ef7b), with subsequent refinements to not have so many irrelevant errors. https://github.com/root-project/root/pull/12418 now adds another case.

**Comment by Axel-Naumann** [comment]
Fixed in master and 6.28/02, thanks for reporting!
